graspologic/simulations/simulations.py

```diff
@@ -795,3 +795,195 @@ def p_from_latent(X, Y=None, rescale=False, loops=True):
         P[P < 0] = 0
         P[P > 1] = 1
     return P
+
+
+def mmsbm(
+    n,
+    p,
+    alpha=None,
+    rng=np.random.default_rng(),
+    directed=False,
+    loops=False,
+    return_labels=False,
+):
+    r"""
+    Samples a graph from Mixed Membership Stochastic Block Model (MMSBM).
+
+    MMSBM produces a graph given the specified block connectivity matrix B,
+    which indicates the probability of connection between nodes based upon
+    their community membership.
+    Each node is assigned a membership vector drawn from Dirichlet distribution
+    with parameter :math:`\alpha`. Each entry of this vector indicates the probability
+    of that node pertaining to each community. Finally, each interaction between
+    nodes is drawn by assigning the community membership of each node according to
+    the probability specified in the mixed membership vector for that node.
+
+
+    Read more in the :ref:`tutorials <simulations_tutorials>`
+
+    Parameters
+    ----------
+    n: int
+        Number of vertices of the graph.
+
+    p: array-like, shape (n_communities, n_communities)
+        Probability of an edge between each of the communities, where p[i, j] indicates
+        the probability of a connection between edges in communities [i, j].
+        0 < p[i, j] < 1 for all i, j.
+
+    alpha: array-like, shape (n_communities,)
+        Concentration parameter :math:`\alpha` of the Dirichlet distribution used to
+        sample the mixed-membership vectors for each node.
+        alpha[i] > 0 for all i.
+
+    rng: numpy.random.Generator, optional (default= numpy.random.default_rng())
+        Numpy Random Generator object to generate sampling from distributions.
```

Comment:
this is a good idea but we haven't implemented in simulations thus far. maybe open an issue, and remove here for now? just seems like it would be inconsistent otherwise

Reply:
im not necessarily opposed to using the `rng.` syntax though

Reply:
can talk about this if you want

Reply:
Yes, I actually utilized the rng. syntax because numpy.random.dirchlet method is quite obsolete and doesn't seem to work very well for small values of alpha(returning NaN while it shouldn't). Here is an [issue](https://github.com/numpy/numpy/issues/5851) from the numpy github pages, where they discuss exactly this problem. Additionally, in the numpy documentation [here](https://numpy.org/doc/stable/reference/random/generated/numpy.random.dirichlet.html) it is noted that: 
> New code should use the dirichlet method of a default_rng() instance instead.

I will also prepare a quick Notebook showing the problem which I will link to this comment. 

I am open to discuss it with you. In case we would prefer switching back I would probably need to modify the Dirichlet method or write my own Dirichlet sampling function according to the [density function](https://en.wikipedia.org/wiki/Dirichlet_distribution) or using some type of stochastic approximation such as [Markov Chain Monte Carlo](https://en.wikipedia.org/wiki/Markov_chain_Monte_Carlo) to approximate the distribution.

Reply:
nvm then